modules/miner/cpuminer.go

```diff
@@ -2,11 +2,18 @@ package miner
 
 import (
 	"time"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
 // threadedMine starts a gothread that does CPU mining. threadedMine is the
 // only function that should be setting the mining flag to true.
 func (m *Miner) threadedMine() {
+	if err := m.tg.Add(); err != nil {
+		return
+	}
+	defer m.tg.Done()
+
```

Comment:
the lock does not protect the threadgroup, so it can be called with or without a lock. Generally these should be the first 4 lines of each tracked function.